gcp-observability/src/main/java/io/grpc/gcp/observability/logging/GcpLogSink.java

```diff
@@ -98,8 +83,9 @@ public GcpLogSink(String destinationProjectId, Map<String, String> locationTags,
   @Override
   public void write(GrpcLogRecord logProto) {
     if (gcpLoggingClient == null) {
-      logger.log(Level.SEVERE, "Attempt to write after GcpLogSink is closed.");
-      return;
+      synchronized (this) {
+        gcpLoggingClient = createLoggingClient();
```

Comment:
Don't you need to check again if `gcpLoggingClient == null`  since the first check was outside the lock?

Reply:
Done